### PR TITLE
Refactor and fix incorrect comment

### PR DESCRIPTION
**Pull request by ethantkoenig**
- Replace the `*User` arguments to `[aA]ccessLevel(..)`, `[hH]asAccess(..)` to `int64` userID arguments to avoid unnecessary loads
- Fix incorrect specifications for `AccessLevel(..)` `HasAccess(..)`
- Add a `getTeamUsersByTeamID(..)` helper function to avoid unnecessary loads

### CODE REVIEW

File: models/access.go [hunk 1]
```diff
@@ -59,41 +59,37 @@ type Access struct {
...
 }
 
-func accessLevel(e Engine, user *User, repo *Repository) (AccessMode, error) {
+func accessLevel(e Engine, userID int64, repo *Repository) (AccessMode, error) {
 	mode := AccessModeNone
 	if !repo.IsPrivate {
 		mode = AccessModeRead
 	}
 
-	if user == nil {
```
Comment (lunny) [inline models/access.go:68]: `if userID == 0` ?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/access.go (hunk 1 above)
Comment (ethantkoenig) [reply models/access.go:68]: Added back.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Pull request merged**